Add documentation for `ttnv3` source of `lorawan-stack-migrate` tool

happyRip opened the pull request:
<!--
Thanks for submitting a pull request. Please fill the template below,
otherwise we will not be able to process this pull request.
-->

#### Summary
<!--
A short summary, referencing related issues:
Closes #0000, References #0000, etc.
-->

Closes #970 

#### Screenshots
<!-- Post a screenshot of your rendered content
NOTE: This section is optional.
-->

![localhost_1313_getting-started_migrating_migration-tool_(Laptop with HiDPI screen)](https://user-images.githubusercontent.com/73070465/211303583-64f22bda-9662-4baf-821d-0c0cdf9cec92.png)

#### Changes
<!-- What are the changes made in this pull request? -->

- Add `ttnv3` source docs page

#### Notes for Reviewers
<!--
NOTE: This section is optional.

Motivate briefly why it is implemented this way, if that deviates from the
implementation proposal in the referenced issues.
- How should your reviewers approach this pull request?
- @mention reviewers with special requests or questions for them
-->

nil

#### Checklist
<!-- Make sure that this pull request is complete. -->

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [x] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

ymgupta commented:
Hi @happyRip,

In addition to the information in this PR, please consider including the below points in the `Configure ttn-lw-migrate` section. 

1. The `Application Server`, `Join Server`, and `Network Server` are all available in the regional clusters. However, the `Identity Server` is only available in the eu1 cluster, i.e., `eu1.cloud.thethings.network`. 
2. Add the instructions on how to get the `TTNV3_CA_FILE` file if anyone want to use this.

adriansmares commented:
Right now this new page sort of overlaps with the TTSCE -> TTSC migration instructions, which is sort of expected but confusing.

I think that we should find a way to integrate the content that we already have there into this page, and the title itself should be more akin to `Migrating between The Things Stack distributions`, with a special step by step (copy paste-able) set of instructions for TTSCE -> TTSC. For reference, `v3` is more of an internal name nowadays than something that people are aware of externally.

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/_index.md:
```diff
@@ -0,0 +1,34 @@
+
+<!--more-->
+
+You can migrate your devices (e.g. from source {{% tts %}} to target {{% tts %}}) using the [migration tool]({{< ref "/getting-started/migrating/migration-tool" >}}), {{% tts %}} [Console]({{< ref "/getting-started/console" >}}) or {{% tts %}} [CLI]({{< ref "/getting-started/cli" >}}), depending on if you want to migrate them with or without their existing session, and how many devices you wish t ...
```
```suggestion
There are multiple methods that you can use to migrate your devices from source {{% tts %}} to target {{% tts %}} depending on if you want to migrate them with or without their existing session, and how many devices you wish to migrate. Migration can be performed using the [migration tool]({{< ref "/getting-started/migrating/migration-tool" >}}), {{% tts %}} [Console]({{< ref "/getting-started/console" >}}) or {{% tts %}} [CLI]({{< ref "/getting-started/cli" >}}.
```

nejraselimovic replied:
Break this sentence into something like this, it's easier to read

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/establish-new-session/_index.md:
```diff
@@ -32,11 +35,23 @@ Since {{% tts %}} Community Edition and {{% tts %}} Cloud are both connected to
+
+{{< tabs/tab "Migration tool" >}}
+
+To migrate devices using the [migration tool]({{< ref "/getting-started/migrating/migration-tool" >}}) without persisting active session use `--ttnv3.no-session` flag, e.g.:
```
```suggestion
To migrate devices using the [migration tool]({{< ref "/getting-started/migrating/migration-tool" >}}) without persisting active session, use the `--ttnv3.no-session` flag:
```

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/establish-new-session/_index.md:
```diff
@@ -70,6 +85,8 @@ Next, you need to import the `device-description.json` file in your {{% tts %}}
 
 {{< /tabs/container >}}
 
+If you migrated devices using the [migration tool]({{< ref "/getting-started/migrating/migration-tool" >}}) the process of migrating OTAA devices ends here. Otherwise follow the last steps below:
```
```suggestion
If you migrated devices using the [migration tool]({{< ref "/getting-started/migrating/migration-tool" >}}) the process of migrating OTAA devices ends here. Otherwise keep following steps below.
```

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/use-migration-tool/_index.md:
```diff
@@ -0,0 +1,89 @@
+weight: 1
+---
+
+This section explains how to migrate end devices between {{% tts %}} distributions.
```
```suggestion
This section explains how to migrate end devices between {{% tts %}} distributions using the migration tool.
```

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/use-migration-tool/_index.md:
```diff
@@ -0,0 +1,89 @@
+
+<!--more-->
+
+First, configure the migration tool with environment variables, or command-line arguments. See `--help` for more details:
```
```suggestion
First, configure the migration tool with environment variables or command-line arguments, like shown below. See `--help` for more details.
```

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/use-migration-tool/_index.md:
```diff
@@ -0,0 +1,89 @@
+
+{{< note >}} The export process will halt if any error occurs. Execute commands with the `--dry-run` flag to verify whether the outcome will be as expected. {{</ note >}}
+
+Then proceed to exporting devices or applications:
```
```suggestion
Then proceed to exporting devices or applications.
```

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/use-migration-tool/_index.md:
```diff
@@ -0,0 +1,89 @@
+
+{{< tabs/tab "Export devices" >}}
+
+To export a single device using it's **DeviceID** (e.g. `my-device`):
```
```suggestion
To export a single device using its **End device ID** (e.g. `my-device`):
```

nejraselimovic replied:
This is just to line it up with names in the Console

nejraselimovic commented on doc/content/getting-started/migrating/migrating-between-tts/use-migration-tool/_index.md:
```diff
@@ -0,0 +1,89 @@
+
+{{< tabs/tab "Export applications" >}}
+
+To export all devices of an application using it's **ApplicationID** (e.g. `my-app-id`):
```
```suggestion
To export all devices of an application using its **Application ID** (e.g. `my-app-id`):
```

nejraselimovic reviewed (commented):
I'm dumping a few comments/suggestions here for readability improvements, so you can apply them while finalizing your draft.

The structure of the whole migration section is currently pretty confusing (e.g. migrating from v3 + migrating between tts sections, migration tool + using migration tool subsection, etc.). If we want to keep the migrating from CE to Cloud section, we should probably put it as a subsection of the migrating between TTS distributions section. We also don't need to explain how to configure the migration tool in 3 different sections, it's creating an overhead and introducing more confusion, and it has been confusing enough already. I didn't put any suggestions regarding this because this is marked as a draft so you might already have an idea on how to fix this :) let me know if you need help

happyRip commented:
I've improved the structure of the docs and finalized the design at least to a review-ready state.

I've applied all of the comments from @nejraselimovic
The migration between TTS distributions section was merged with ttnv3 migrate tool section.
I added a table with links to sections of each source in getting-started/migration-tool section.

In `CONTRIBUTING.md` it is specified to use `$` before any bash commands, but most of the docs I came by didn't do this, so I removed them to match this style, this can be done the other way around if needed.

nejraselimovic commented:
> I've improved the structure of the docs and finalized the design at least to a review-ready state.
> 
> I've applied all of the comments from @nejraselimovic The migration between TTS distributions section was merged with ttnv3 migrate tool section. I added a table with links to sections of each source in getting-started/migration-tool section.
> 
> In `CONTRIBUTING.md` it is specified to use `$` before any bash commands, but most of the docs I came by didn't do this, so I removed them to match this style, this can be done the other way around if needed.

Thanks for doing this, we have some great content there! This is a great progress, but I do have a small suggestion before we finalize this. See the attachment below for the reference. The **Migrate using the Migration Tool** subsection is bugging me, as I think it's creating a confusion - we'll have a loop between sections **Migration Tool** - **Migrate using the Migration Tool** - **Migrate without Persisting Active Session**/**Migrate Active Device Session** and users won't be able to easily understand what's going on.

My suggestion to do one of the following:
1. Remove the **Migrate using the Migration Tool** subsection and transfer that info to **Migration Tool** (environment variables configuration) and **Migrate without Persisting Active Session**/**Migrate Active Device Session** (commands for device and app migration (basically, we already have this in the Migration Tool tab together with CLI and Console)).
2. Reorganize **Migrate using the Migration Tool**, **Migrate without Persisting Active Session** and **Migrate Active Device Session** similar to what we have in **Migrating End Devices from V2**, i.e. define 3 subsections **Migrate using the Migration Tool**, **Migrate using the Console** and **Migrate using the CLI**, and then explain how to do migration with and without persisting active device session in each one of those.

I think one of these suggestions would make it easier to follow, but let me know your thoughts as well.

![Screenshot from 2022-12-06 20-24-59](https://user-images.githubusercontent.com/65215579/206003656-c9b7bb83-00c8-4d1a-94f6-4f88c0b204fb.png)

happyRip commented:
I decided to try and follow

> 1. Remove the Migrate using the Migration Tool subsection and transfer that info to Migration Tool (environment variables configuration) and Migrate without Persisting Active Session/Migrate Active Device Session (commands for device and app migration (basically, we already have this in the Migration Tool tab together with CLI and Console)).

V2 instructions were kept in the previous location for 2 reasons:
- AFAIK it is depreciated (if there is an option to mark it depreciated for migration tool and not the stack than I could add that)
- the V2 section is not only about using the migration tool

This way in the **Migration tool** section there is TTS and ChirpStack for now

As a benefit on the bottom of instructions on installing the migration tool there is a list of subpages with instructions on using the pages, so the table I made before may be obsolete, not sure
<img width="1059" alt="image" src="https://user-images.githubusercontent.com/73070465/208737335-2de40446-cda8-4832-8da7-009a53850ec0.png">

The name of the section **Migrating End Devices from The Things Stack** doesn't seem perfect to me, but I couldn't think of a better one.

nejraselimovic commented:
@happyRip I pushed some changes, can you check them out? You'll notice I changed some parts to distinguish the process of **exporting devices** and the process of actually **migrating devices**, which I think will make things clearer for readers. Other changes are mostly for improving readability. I'm also planning on removing the docs about migration from V2 as Rish said that now it is safe to remove them since V2 was shut down a year ago, but can handle that in a separate PR.

happyRip commented:
I've checked-out the changes and I think distinguishing between **migrating** and **exporting** works wonders here. Thank you for improving the content as well.

This branch got a bit old, I'm going to perform a rebase and resolve merge conflicts.

adriansmares commented on doc/content/getting-started/migrating/migrating-between-tts-distributions/establish-new-session/_index.md:
```diff
@@ -1,52 +1,73 @@
 ### Note on temporarily preserving uplink traffic on {{% tts %}} Community Edition
 
-In this section, we consider migrating your devices in cases when you don't want to migrate your gateway from {{% tts %}} Community Edition to {{% tts %}} Cloud, or when your gateway is inaccessible for this migration. In those cases, devices are migrated from {{% tts %}} Community Edition to {{% tts %}} Cloud without persisting their active session, i.e. devices need to establish a new session w ...
+In this section, we consider migrating your devices in cases when you don't want to migrate your gateway from {{% tts %}} Community Edition to {{% tts %}} Cloud, or when your gateway is inaccessible for this migration. In those cases, devices are migrated from {{% tts %}} Community Edition to {{% tts %}} Cloud without persisting their active session, i.e. devices need to establish a new session w ...
```
`in order for a gateway`.
